shared/src/commonMain/kotlin/co/touchlab/kampkit/models/BreedModel.kt

```diff
@@ -31,20 +32,36 @@ class BreedModel : KoinComponent {
         ensureNeverFrozen()
     }
 
-    fun getBreeds(): Flow<DataState<ItemDataSummary>> = flow {
+    fun getBreeds(forced: Boolean = false): Flow<DataState<ItemDataSummary>> = flow {
         emit(DataState.Loading)
         val currentTimeMS = clock.now().toEpochMilliseconds()
         val stale = isBreedListStale(currentTimeMS)
         val networkBreedDataState: DataState<ItemDataSummary>
-        if (stale) {
+        if (stale || forced) {
             networkBreedDataState = getBreedsFromNetwork(currentTimeMS)
             when (networkBreedDataState) {
                 DataState.Empty -> {
                     // Pass the empty response along
                     emit(networkBreedDataState)
                 }
                 is DataState.Success -> {
-                    dbHelper.insertBreeds(networkBreedDataState.data.allItems.map { it.name })
+                    val breeds = mutableListOf<Breed>()
+                    val breedsInCacheItemDataSummary = getBreedsFromCache().first()
+                    if (breedsInCacheItemDataSummary is DataState.Success) {
+                        breeds.addAll(breedsInCacheItemDataSummary.data.allItems)
+                    }
+                    dbHelper.insertBreeds(
+                        networkBreedDataState.data.allItems.map { breed ->
+                            var toInsert: Breed = breed
+                            val cacheBreedIndex = breeds.indexOf(breed)
+                            if (cacheBreedIndex > -1) {
+                                toInsert = breeds[cacheBreedIndex]
+                                log.d { breed.toString() }
+                                log.d { toInsert.toString() }
+                            }
+                            return@map toInsert
+                        }
+                    )
```

Comment:
I don't follow all the changes here. We used to pass everything from network into sql, and update records that were duplicates of things already in the db. Now we grab all items from the db, prefer previously cached results to new ones from the web call, and then ignore them on reinsert. It feels like this is doing more work than it needs to and I don't understand what this change enables that wasn't happening before.

Reply:
The reason for going through all this is because replacing the breeds with those from the network doesn't preserve whether they are liked or not. So I'm doing this comparison, seeing if they're in the DB, and if they are, replacing them w/ the one already there, because it will have the "liked" boolean instead of defaulting to liked=false.